source/deploy-strategies.rst

```diff
@@ -0,0 +1,162 @@
+Choosing a deploy strategy
+==========================
+
+Different projects at Caktus use different tools and strategies
+for provisioning and deploying. This is partly for historical reasons,
+but also because different projects have different requirements.
+
+Different approaches
+~~~~~~~~~~~~~~~~~~~~
+
+Simple Fabric
+-------------
+
+For some of our earliest projects, we just did everything in a
+`Fabric <http://www.fabfile.org/>`_
+script, without the assistance of any other tools. That's error-prone
+and a lot of work. We would never do that for a new project, and I'm
+not sure any project we still maintain does that.
+
+FabulAWS
+--------
+
+`FabulAWS <https://fabulaws.readthedocs.io/en/latest/>`_
+is a Caktus-developed (primarily Tobias) framework
+for deploying autoscaling, redundant projects to Amazon Web Services.
+Its primary advantage is built-in support for autoscaling: adding and
+removing servers as the load on the site changes.
+
+Two projects use FabulAWS that I'm aware of: CCSR and OpenDebates. Both
+can experience heavy loads at times, with significant load changes over
+time.
+
+These days,
+`Elastic Beanstalk <https://aws.amazon.com/elasticbeanstalk/>`_
+is a credible alternative to FabuLAWS.
+It also supports autoscaling, using the same underlying AWS mechanisms
+as FabulAWS, and we're not responsible for maintaining
+it ourselves.
+
+Both FabulAWS and Elastic Beanstalk are tied to AWS.
+
+Margarita
+---------
+
+Most of our current client projects use a Caktus project called
+`Margarita <http://caktus.github.io/developer-documentation/margarita/margarita.html>`_
+for deployment, which uses the
+`Salt <https://saltstack.com/>`_ configuration management system
+to deploy the site.
+
+This has worked reasonably well (well enough that we keep using it), but
+over time we've discovered a number of shortcomings:
+
+* Difficulty debugging deploy problems
+* Fragility of Salt updates - many new updates seem to introduce new
+  bugs, and the next update that fixes those might have its own new
+  bugs. It can be tricky figuring out a version of Salt that works.
+* Resource usage - the Salt master process seems to grow over time to
+  use a lot of memory on the server, even though it shouldn't be doing
+  anything at all when we're not deploying. This ties up memory our
+  site could be using.
+
+We are therefore moving away from Margarita for new projects.
```

Comment:
This wording feels too strong given the relatively soft recommendation for Tequila. Currently one of those two should be the default with other options (Beanstalk/ECS/Dokku) being potentially better suited depending on the client/use case.